<launch>

    <arg name="wait_for_arm" default="true"/>
    <arg name="use_gripper" default="false"/>
    <arg name="land_separate" default="false"/>
    <arg name="record" default="false"/>
    <arg if="$(arg record)" name="prefix"/>

    <node pkg="softdrone_core" type="gain_tuning_node" name="gain_tuning_node" output="screen">
        <remap from="state" to="/mavros/state"/>
        <remap from="pose" to="/mavros/local_position/pose"/>
        <remap from="target" to="/mavros/setpoint_raw/local"/>
        <remap from="lengths" to="/gripper/lengths"/>
        <param name="wait_for_arm" value="$(arg wait_for_arm)"/>
        <param name="dist_threshold" value="0.3"/>
        <param name="hover_duration" value="6.0"/>
        <param name="total_time" value="2.75"/>
        <param name="land_separate" value="$(arg land_separate)"/>
        <param name="start_settle_duration" value="5.0"/>
        <rosparam param="start_position">[-1.0, -1.5, 2.5]</rosparam>
        <rosparam param="tendon_target">[2.0, -2.5, 2.0]</rosparam>
        <rosparam param="end_velocity">[0.0, 0.0, -0.1]</rosparam>
        <rosparam param="land_position">[2.75, -3.6, 2.5]</rosparam>
    </node>

    <node if="$(arg use_gripper)" pkg="softdrone_core" type="gripper_node" name="gripper" output="screen">
        <remap from="lengths" to="/gripper/lengths"/>
    </node>

    <node if="$(arg record)" pkg="rosbag" type="record" name="recorder" args="-a -o $(arg prefix)"/>

</launch>
